syft/frameworks/torch/tensors/interpreters/bfv.py

```diff
@@ -0,0 +1,301 @@
+import syft as sy
+import numpy as np
+import torch as th
+
+from syft.generic.abstract.tensor import AbstractTensor
+from syft.generic.frameworks.hook.hook_args import (
+    get_child,
+    register_backward_func,
+    register_forward_func,
+    register_type_rule,
+    one,
+)
+from syft.generic.frameworks.overload import overloaded
+from syft.workers.abstract import AbstractWorker
+
+from syft.frameworks.torch.he.fv.decryptor import Decryptor
+from syft.frameworks.torch.he.fv.encryptor import Encryptor
+from syft.frameworks.torch.he.fv.integer_encoder import IntegerEncoder
+from syft.frameworks.torch.he.fv.evaluator import Evaluator
+
+
+class BFVTensor(AbstractTensor):
+    def __init__(self, **kwargs):
+        """Initializes a BFVTensor.
+
+        Args:
+            owner: An optional BaseWorker object to specify the worker on which
+                the tensor is located.
+            id: An optional string or integer id of the BFVTensor.
+        """
+        super().__init__(**kwargs)
+        self.context = None
+        self.encryptor = None
+        self.decryptor = None
+        self.encoder = None
+        self.evaluator = None
+
+    def encrypt(self, key, context):
+        self.context = context
+        self.prepareEncryptor(key)
+
+        output = BFVTensor()
+        output.child = self.child
+        inputs = self.child.flatten().tolist()
+        new_child = [self.encryptor.encrypt(self.encoder.encode(x)) for x in inputs]
+
+        data = np.array(new_child).reshape(self.child.shape)
+        self.child = data
+        return output
+
+    def decrypt(self, private_key):
+        self.prepareDecryptor(private_key)
+
+        inputs = self.child.flatten().tolist()
+        new_child = [self.encoder.decode(self.decryptor.decrypt(x)) for x in inputs]
+        return th.tensor(new_child).view(*self.child.shape)
+
+    def __add__(self, *args, **kwargs):
+        self.prepareEvaluator()
+
+        if not args[0].shape == self.child.shape:
+            raise ValueError(
+                f"Cannot add tensors with different shape {args[0].shape} , {self.child.shape}"
+            )
+
+        args = args[0].flatten().tolist()
+
+        # if integer values; transform it to plaintext
+        if isinstance(args[0], int):
+            args = [self.encoder.encode(x) for x in args]
+
+        child = self.child.flatten().tolist()
+
+        data = [self.evaluator.add(x, y) for x, y in zip(child, args)]
+        data = np.array(data).reshape(self.child.shape)
```

Comment:
Have you tried to implement the `__radd__`, `__rsub__` methods so that it also can operate when it's the right operand. However, I'm not sure this is how we should do it in PySyft.